packages/lab/src/radio-button/RadioButton.tsx

```diff
@@ -1,86 +1,156 @@
 
 const withBaseName = makePrefixer("saltRadioButton");
 
 export interface RadioButtonProps
-  extends Omit<HTMLAttributes<HTMLDivElement>, "onChange"> {
+  extends Omit<
+    HTMLAttributes<HTMLLabelElement>,
+    "onChange" | "onBlur" | "onFocus"
+  > {
+  /**
+   * Set the default selected radio button in the group
+   */
   checked?: boolean;
   className?: string;
+  /**
+   * Prop used to set the styling
+   */
   disabled?: boolean;
   /**
-   * Custom icon component
+   * Prop used to set the styling
+   */
+  error?: boolean;
+  id?: string;
+  /**
+   * Props to be passed to the radio input
+   */
+  inputProps?: Partial<InputHTMLAttributes<HTMLInputElement>>;
+  /**
+   * The label to be shown next to the radio icon
    */
-  icon?: ComponentType<RadioButtonIconProps>;
+  label?: ReactNode;
   /**
-   * The label to be shown next to the radio
+   * Name of the radio group
    */
-  label?: ControlLabelProps["label"];
-  LabelProps?: Partial<ControlLabelProps>;
   name?: string;
+  /**
+   * Callback for blur event
+   */
+  onBlur?: FocusEventHandler<HTMLSpanElement>;
+  /**
+   * Callback for change event
+   */
   onChange?: ChangeEventHandler<HTMLInputElement>;
+  /**
+   * Callback for focus event
+   */
+  onFocus?: FocusEventHandler<HTMLSpanElement>;
+  /**
+   * Value of radio button
+   */
   value?: string;
 }
 
 export const RadioButton = forwardRef<HTMLLabelElement, RadioButtonProps>(
   function RadioButton(props, ref) {
     const {
-      checked,
+      checked: checkedProp,
       className,
-      icon,
       disabled,
+      error,
+      id: idProp,
+      inputProps,
       label,
-      LabelProps,
-      value,
+      name: nameProp,
+      onFocus,
+      onBlur,
       onChange,
+      value,
       ...rest
     } = props;
 
+    const id = idProp || useId();
+    const radioGroup = useRadioGroup();
+
+    let radioGroupChecked = checkedProp;
+    let name = nameProp;
+
+    if (Object.keys(radioGroup).length) {
+      if (typeof radioGroupChecked === "undefined") {
+        radioGroupChecked = radioGroup.value === value;
+      }
+      if (typeof name === "undefined") {
+        name = radioGroup.name;
+      }
+    }
+
+    const [checked, setCheckedState] = useControlled({
+      controlled: radioGroupChecked,
+      default: Boolean(checkedProp),
+      name: "RadioBase",
+      state: "checked",
+    });
+
+    const handleFocus: FocusEventHandler<HTMLElement> = (event) => {
+      onFocus && onFocus(event);
+    };
+
+    const handleBlur: FocusEventHandler<HTMLElement> = (event) => {
+      onBlur && onBlur(event);
+    };
+
+    const handleInputChange: ChangeEventHandler<HTMLInputElement> = (event) => {
+      const newChecked = event.target.checked;
+      setCheckedState(newChecked);
+
+      onChange && onChange(event);
+
+      if (Object.keys(radioGroup).length) {
+        radioGroup.onChange && radioGroup.onChange(event);
+      }
+    };
+
     return (
-      <div
+      <label
```

Comment:
Just curious, if used without a label is an axe error produced?

Reply:
I think we should add an example for it either way

Reply:
It wouldn't be accessible. 
I've added the missing checkAccessibility test on RadioGroup